op-batcher/batcher/driver.go

```diff
@@ -773,10 +777,22 @@ func (l *BatchSubmitter) recordFailedTx(id txID, err error) {
 	l.state.TxFailed(id)
 }
 
-func (l *BatchSubmitter) recordConfirmedTx(id txID, receipt *types.Receipt) {
+func (l *BatchSubmitter) recordConfirmedTx(id txID, receipt *types.Receipt) error {
 	l.Log.Info("Transaction confirmed", logFields(id, receipt)...)
-	l1block := eth.ReceiptBlockID(receipt)
-	l.state.TxConfirmed(id, l1block)
+	tctx, cancel := context.WithTimeout(l.shutdownCtx, l.Config.NetworkTimeout)
+	defer cancel()
+	header, err := l.L1Client.HeaderByNumber(tctx, receipt.BlockNumber)
+	if err != nil {
+		return err
+	}
```

Comment:
There's an extra network call here to get the block timestamp, which we need to compare with the holocene time. 

It might be good to put some retry logic in here, `op-service` exposes some utilities for this. 

Currently if this network call fails, the channel timeout checks are not performed, even though in principle they still could be because the block number is still known.

Reply:
See my review, I think we should try indefinitely to get the header. It's important for future batcher operation that we correctly act on receipts.